src/App.tsx

```diff
@@ -248,6 +250,7 @@ function App() {
         setIsSettingsModalOpen={setIsSettingsModalOpen}
       />
       <div className="pt-2 px-1 pb-8 md:max-w-7xl w-full mx-auto sm:px-6 lg:px-8 flex flex-col grow">
+        <Solution isVisible={DEBUG_SHOW_SOLUTION} solution={solution} />
```

Comment:
Nit: I prefer this since it makes the `Solution` more single responsibility.
```suggestion
        {DEBUG_SHOW_SOLUTION && <Solution solution={solution} />}
```

Reply:
Ok, again, me learning TypeScript / React.